metron-analytics/metron-profiler-storm/src/main/java/org/apache/metron/hbase/bolt/HBaseBolt.java

```diff
@@ -76,60 +77,67 @@
   protected HBaseMapper mapper;
 
   /**
-   * The name of the class that should be used as a table provider.
-   *
-   * <p>Defaults to 'org.apache.metron.hbase.HTableProvider'.
+   * Defines when a batch needs flushed.
    */
-  protected String tableProviderClazzName = "org.apache.metron.hbase.HTableProvider";
+  private BatchHelper batchHelper;
 
   /**
-   * The TableProvider
-   * May be loaded from tableProviderClazzName or provided
+   * Establishes a connection to HBase.
    */
-  protected TableProvider tableProvider;
+  private HBaseConnectionFactory connectionFactory;
+
+  /**
+   * Creates the {@link HBaseTableClient} used by this bolt.
+   */
+  private HBaseClientFactory hBaseClientFactory;
+
+  /**
+   * Used to write to HBase.
+   */
+  protected transient HBaseClient hbaseClient;
 
-  private BatchHelper batchHelper;
   protected OutputCollector collector;
-  protected transient LegacyHBaseClient hbaseClient;
 
   public HBaseBolt(String tableName, HBaseMapper mapper) {
     this.tableName = tableName;
     this.mapper = mapper;
+    this.connectionFactory = new HBaseConnectionFactory();
```

Comment:
I'm a little foggy on the difference between these two factories, same as my note in the config.